apps/console/src/features/applications/utils/application-management-utils.ts

```diff
@@ -339,8 +339,17 @@ export class ApplicationManagementUtils {
 
         let callbackURL = urls.replace(/['"]+/g, "");
 
+        const regexChars = /[.*+?^${}()|[\]\\]/g; // Regex that matches special characters.
         if (callbackURL.split(",").length > 1) {
-            callbackURL = "regexp=(" + callbackURL.split(",").join("|") + ")";
+            callbackURL = callbackURL.split(",").map((url) => {
+                if (!/\\/.test(url) && regexChars.test(url)) {
+                    url = url.replace(regexChars, '\\$&'); // Escape the special character.
+                }
+                return url;
```

Comment:
Add newline before return statement.
```suggestion

                return url;
```